src/EventSubscriber/AuthorCorporateBodyExtractDataSubscriber.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content\EventSubscriber;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\oe_content\AuthorSubEntitySubscriberBase;
+use Drupal\oe_content\Event\AuthorExtractDataEvent;
+use Drupal\rdf_skos\Entity\ConceptInterface;
+
+/**
+ * Event subscriber for extracting data for "Corporate body" author bundles.
+ */
+class AuthorCorporateBodyExtractDataSubscriber extends AuthorSubEntitySubscriberBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function applies(ContentEntityInterface $entity): bool {
+    return $entity->getEntityTypeId() === 'oe_author' && $entity->bundle() === 'oe_corporate_body';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function extractLinks(AuthorExtractDataEvent $event): void {
+    // Load referenced entities.
+    $entities = $event->getEntity()->referencedEntities();
+
+    $links = [];
+    foreach ($entities as $entity) {
+      if ($entity instanceof ConceptInterface) {
+        $event->addCacheableDependency($entity);
```

Comment:
Currently SKOS concept term labels are shown in english language for translated pages. Can we translate SKOS terms using `$entity = \Drupal::service('entity.repository')->getTranslationFromContext($entity);`?

Reply:
Fixed. Please pay attention during testing that it requires to make the Concept entity type translatable.

Reply:
\Drupal::service has to be replaced on the container injection invocation.